website/docs/basics/autowiring.md

```diff
@@ -6,3 +6,213 @@ sidebar_label: Autowiring
+
+Sounds confusing, right?
+
+Let's us provide you with some code examples:
+
+**Structure:**
+- src
+	- Query
+		- Documents
+			- QueryDocumentsInterface.class
+			- QueryDocuments.class
+	- Rest
+		- Routes
+			- DocumentsRoute.class
+
+
+**src/Query/Documents/QueryDocumentsInterface.class**
+```js
+/**
+ * The file that is an Query for Documents class interface.
+ *
+ * @package ProjectNamespace\Query\Documents;
+ */
+
+declare(strict_types=1);
+
+namespace ProjectNamespace\Query\Documents;
+
+/**
+ * Interface for a QueryDocumentsInterface
+ */
+interface QueryDocumentsInterface
+{
+
+  /**
+   * Get structured documents data for documents section.
+   *
+   * @param int $id Id for data.
+   *
+   * @return string
+   */
+  public function getDocumentsData(int $id = 0): string;
+}
+```
+
+**src/Query/Documents/QueryDocuments.class**
+```js
+/**
+ * The file that is an Query of Documents class.
+ *
+ * @package ProjectNamespace\Query\Documents
+ */
+
+declare(strict_types=1);
+
+namespace ProjectNamespace\Query\Documents;
+
+use ProjectNamespaceVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * QueryDocuments class.
+ */
+class QueryDocuments implements ServiceInterface, QueryDocumentsInterface
+{
+    /**
+   * Get structured documents data for documents section.
+   *
+   * @param int $id Id for data.
+   *
+   * @return string
+   */
+  public function getDocumentsData(int $id = 0): string
+  {
+    ... here goes the implementation of getDocumentsData method.
+  }
+}
+```
+
+**src/Rest/Routes/DocumentsRoute.class**
+```js
+/**
+ * The class register route for DocumentsRoute endpoint
+ *
+ * @package ProjectNamespace\Rest\Routes
+ */
+
+declare(strict_types=1);
+
+namespace ProjectNamespace\Rest\Routes;
+
+use ProjectNamespace\Query\Documents\QueryDocumentsInterface;
+use ProjectNamespaceVendor\EightshiftLibs\Rest\Routes\AbstractRoute;
+use ProjectNamespaceVendor\EightshiftLibs\Rest\CallableRouteInterface;
+
+/**
+ * Class DocumentsRoute
+ */
+class DocumentsRoute extends AbstractRoute implements CallableRouteInterface
+{
+
+  /**
+   * Instance variable of documents data.
+   *
+   * @var QueryDocumentsInterface
+   */
+  protected $documentsData;
+
+  /**
+   * Create a new instance.
+   *
+   * @param QueryDocumentsInterface $queryDocuments Inject documentsData which holds data for documents.
+   */
+  public function __construct(QueryDocumentsInterface $queryDocuments)
+  {
+    $this->documentsData = $queryDocuments;
+  }
+
+  ...here goes the rest of the DocumentsRoute implementation.
+}
+```
+
+An this is it, autowiring knows how to resolve this setup, and you are all set.
+
+### What if I have to mock or do some weird stuff in my DI?
+
+Sometimes automatization can save you, but in the end, eventually, you need something custom, we got you covered.
+
+We can think of these scenarios:
+- You have to write tests and manually provide mocked classes.
+- You need to do some weird stuff in your DI.
```

Comment:
```suggestion
```

What is weird stuff? I never did anything weird in my DIC 😂 

We can remove anything that is overly vague 😉